tests/TestCase/Php7CompareTrait.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         1.1.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Bake\Test\TestCase;
+
+use Cake\TestSuite\StringCompareTrait;
+
+trait Php7CompareTrait
+{
+    use StringCompareTrait {
+        assertSameAsFile as assertSameAsFileBase;
+    }
+
+    /**
+     * Compare the result to the contents of the file based upon which
+     * version PHP is being used.
+     *
+     * The env variable UPDATE_TEST_COMPARISON_FILES will not generate "php7"
```

Comment:
@markstory The @method tags use type hinting on the return value which is PHP7 only.